sdk/iot/core/tests/cmocka/az_iot_telemetry_tests.c

```diff
@@ -0,0 +1,301 @@
+                       .device_id = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_ID),
+                       .options = TEST_VALID_OPTIONS_BOTH } };
+static const az_iot_hub_client g_test_valid_client_with_options_module_id
+    = { ._internal = { .iot_hub_hostname = AZ_SPAN_LITERAL_FROM_STR(TEST_FQDN),
+                       .device_id = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_ID),
+                       .options = TEST_VALID_OPTIONS_MODULE_ID } };
+static const az_iot_hub_client g_test_valid_client_with_options_user_agent
+    = { ._internal = { .iot_hub_hostname = AZ_SPAN_LITERAL_FROM_STR(TEST_FQDN),
+                       .device_id = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_ID),
+                       .options = TEST_VALID_OPTIONS_USER_AGENT } };
+
+void az_iot_hub_client_telemetry_publish_topic_get_NULL_client_fails(void** state)
+{
+  (void)state;
+
+  uint8_t mqtt_topic_buf[TEST_MQTT_SPAN_BUFFER_SIZE];
+
+  az_span mqtt_topic = AZ_SPAN_LITERAL_FROM_BUFFER(mqtt_topic_buf);
+
+  assert_true(
+      az_iot_hub_client_telemetry_publish_topic_get(NULL, NULL, mqtt_topic, &mqtt_topic)
+      == AZ_ERROR_ARG);
+}
+
+void az_iot_hub_client_telemetry_publish_topic_get_NULL_mqtt_topic_fails(void** state)
+{
+  (void)state;
+  az_span null_mqtt_topic = TEST_NULL_PTR_SPAN;
+
+  assert_true(
+      az_iot_hub_client_telemetry_publish_topic_get(
+          &g_test_valid_client_no_options, NULL, null_mqtt_topic, &null_mqtt_topic)
+      == AZ_ERROR_ARG);
+}
+
+void az_iot_hub_client_telemetry_publish_topic_get_NULL_out_mqtt_topic_fails(void** state)
+{
+  (void)state;
+  uint8_t mqtt_topic_buf[TEST_MQTT_SPAN_BUFFER_SIZE];
+
+  az_span mqtt_topic = AZ_SPAN_LITERAL_FROM_BUFFER(mqtt_topic_buf);
+
+  assert_true(
+      az_iot_hub_client_telemetry_publish_topic_get(
+          &g_test_valid_client_no_options, NULL, mqtt_topic, NULL)
+      == AZ_ERROR_ARG);
+}
+
+void az_iot_hub_client_telemetry_publish_topic_get_no_options_no_params_succeed(void** state)
+{
+  (void)state;
+
+  uint8_t mqtt_topic_buf[sizeof(g_test_correct_topic_no_options_no_params)];
+
+  az_span mqtt_topic = AZ_SPAN_LITERAL_FROM_BUFFER(mqtt_topic_buf);
+
+  assert_true(
+      az_iot_hub_client_telemetry_publish_topic_get(
+          &g_test_valid_client_no_options, NULL, mqtt_topic, &mqtt_topic)
+      == AZ_OK);
+  assert_true(
+      strncmp(
+          g_test_correct_topic_no_options_no_params,
+          (char*)az_span_ptr(mqtt_topic),
+          sizeof(g_test_correct_topic_no_options_no_params))
+      == 0);
+}
+
+void az_iot_hub_client_telemetry_publish_topic_get_with_options_no_params_succeed(void** state)
+{
+  (void)state;
+
+  uint8_t mqtt_topic_buf[sizeof(g_test_correct_topic_with_options_no_params)];
+
+  az_span mqtt_topic = AZ_SPAN_LITERAL_FROM_BUFFER(mqtt_topic_buf);
+
+  assert_true(
+      az_iot_hub_client_telemetry_publish_topic_get(
+          &g_test_valid_client_with_options_both, NULL, mqtt_topic, &mqtt_topic)
+      == AZ_OK);
+  assert_true(
```

Comment:
>assert_true [](start = 2, length = 11)

do they have assert_eq? I like that then this because that print the values also that we are checking.

Reply:
In case of failure.

---
In reply to: [389175305](https://github.com/CIPop/azure-sdk-for-c/pull/1#discussion_r389175305) [](ancestors = 389175305)

Reply:
ah yes they have an `assert_string_equal` i can try using